src/main/java/commandparser/StringParser.java

```diff
@@ -0,0 +1,131 @@
+package commandparser;
+
+import data.TaskManagerException;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class StringParser {
+    private static final DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+
+    /**
+     * Parses a date string into a LocalDate object.
+     *
+     * @param dateString the string representing the date
+     * @return the LocalDate object parsed from the string
+     * @throws TaskManagerException if the date string is not in the expected format
+     */
+    public static LocalDate parseDate(String dateString) throws TaskManagerException {
+        try {
+            return LocalDate.parse(dateString, dateFormatter);
+        } catch (DateTimeParseException e) {
+            throw new TaskManagerException("Invalid date format. Please use the format dd/MM/yyyy.");
+        }
+    }
+
+    /**
+     * Parses a task index string into an integer.
+     *
+     * @param indexString the string representing the task index
+     * @return the integer value of the task index
+     * @throws TaskManagerException if the task index string is not a valid integer
+     */
+    public static int parseTaskIndex(String indexString) throws TaskManagerException {
+        try {
+            return Integer.parseInt(indexString);
+        } catch (NumberFormatException e) {
+            throw new TaskManagerException("Invalid task index. Please enter a valid integer.");
+        }
+    }
+
+    /**
+     * Parses a task description string.
+     *
+     * @param descriptionString the string representing the task description
+     * @return the parsed task description string
+     */
+    public static String parseTaskDescription(String descriptionString) {
+        return descriptionString.trim();
+    }
+
+    /**
+     * Parses a priority level string.
+     *
+     * @param priorityString the string representing the priority level
+     * @return the parsed priority level string
+     * @throws TaskManagerException if the priority level string is not valid
+     */
+    public static String parsePriorityLevel(String priorityString) throws TaskManagerException {
+        String trimmedPriority = priorityString.trim().toUpperCase();
+        if (trimmedPriority.equals("HIGH") || trimmedPriority.equals("MEDIUM") || trimmedPriority.equals("LOW")) {
+            return trimmedPriority;
+        } else {
+            throw new TaskManagerException("Invalid priority level. Please use 'high', 'medium', or 'low'.");
+        }
+    }
+
+    /**
+     * Validates the format of an "add" command.
+     *
+     * @param parts the array of command parts
+     * @throws TaskManagerException if the command format is invalid
+     */
+    public static void validateAddCommand(String[] parts) throws TaskManagerException {
+        if (parts.length < 4) {
```

Comment:
Is there a reason why this command checks for < 4 while lines 100 and 113 check for exact equality?

Reply:
Good catch! On second thought, the inequality check is more flexible and robust hence I will change to it.